tests/components/amberelectric/test_config_flow.py

```diff
@@ -0,0 +1,148 @@
+"""Tests for the Amber config flow."""
+
+from typing import Generator
+from unittest.mock import Mock, patch
+
+from amberelectric import ApiException
+from amberelectric.model.site import Site
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.components.amberelectric.const import (
+    CONF_SITE_ID,
+    CONF_SITE_NAME,
+    CONF_SITE_NMI,
+    DOMAIN,
+)
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_API_TOKEN
+from homeassistant.core import HomeAssistant
+
+API_KEY = "psk_123456789"
+
+
+@pytest.fixture(name="invalid_key_api")
+def mock_invalid_key_api() -> Generator:
+    """Return an authentication error."""
+    instance = Mock()
+    instance.get_sites.side_effect = ApiException(status=403)
+
+    with patch("amberelectric.api.AmberApi.create", return_value=instance):
+        yield instance
+
+
+@pytest.fixture(name="api_error")
+def mock_api_error() -> Generator:
+    """Return an authentication error."""
+    instance = Mock()
+    instance.get_sites.side_effect = ApiException(status=500)
+
+    with patch("amberelectric.api.AmberApi.create", return_value=instance):
+        yield instance
+
+
+@pytest.fixture(name="single_site_api")
+def mock_single_site_api() -> Generator:
+    """Return a single site."""
+    instance = Mock()
+    site = Site("01FG0AGP818PXK0DWHXJRRT2DH", "11111111111", [])
+    instance.get_sites.return_value = [site]
+
+    with patch("amberelectric.api.AmberApi.create", return_value=instance):
+        yield instance
+
+
+@pytest.fixture(name="no_site_api")
+def mock_no_site_api() -> Generator:
+    """Return no site."""
+    instance = Mock()
+    instance.get_sites.return_value = []
+
+    with patch("amberelectric.api.AmberApi.create", return_value=instance):
+        yield instance
+
+
+async def test_single_site(hass: HomeAssistant, single_site_api: Mock) -> None:
+    """Test single site."""
+    initial_result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert initial_result.get("type") == data_entry_flow.RESULT_TYPE_FORM
+    assert initial_result.get("step_id") == "user"
+
+    # Test filling in API key
+    enter_api_key_result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={CONF_API_TOKEN: API_KEY},
+    )
+    assert enter_api_key_result.get("type") == data_entry_flow.RESULT_TYPE_FORM
+    assert enter_api_key_result.get("step_id") == "site"
+
+    select_site_result = await hass.config_entries.flow.async_configure(
```

Comment:
We normally patch the integration setup function on all create entry results to avoid setting up the whole integration. See the scaffold test template for example.